File: Algorithum/Code_Help_By_Babbar/01_recursion/lec31/01_factorial.js
const factNum = 500;

function factorial(n) {
  if (n === 0) return 1;
  return n * factorial(n - 1);
}

console.log(factorial(factNum));

// 5! = 5 * (5 - 1)!
// n! = n * (n - 1)!
// f(n) = n * f(n -1)

File: Code_Help_By_Babbar/31_recursion/01_factorial.cpp
#include <iostream>
using namespace std;

int factorial(int n)
{
    if (n == 0)
        return 1;

    return n * factorial(n - 1);
}

int main()
{
    int n;
    cout << "Enter factorial no : ";
    cin >> n;

    cout << "Factorial is : " << factorial(n);
}